AsyncDisplayKit/ASMapNode.mm

```diff
@@ -0,0 +1,210 @@
+{
+  ASDN::MutexLocker l(_propertyLock);
+  return _liveMap;
+}
+
+- (void)setLiveMap:(BOOL)liveMap
+{
+  ASDN::MutexLocker l(_propertyLock);
+  if (liveMap == _liveMap) {
+    return;
+  }
+  _liveMap = liveMap;
+  liveMap ? [self addLiveMap] : [self removeLiveMap];
+}
+
+
+- (BOOL)needsMapReloadOnBoundsChange
+{
+  ASDN::MutexLocker l(_propertyLock);
+  return _needsMapReloadOnBoundsChange;
+}
+
+- (void)setNeedsMapReloadOnBoundsChange:(BOOL)needsMapReloadOnBoundsChange
+{
+  ASDN::MutexLocker l(_propertyLock);
+  _needsMapReloadOnBoundsChange = needsMapReloadOnBoundsChange;
+}
+
+- (void)fetchData
+{
+  [super fetchData];
+  if (_liveMap && !_mapNode) {
+    [self addLiveMap];
+  }
+  else {
+    [self setUpSnapshotter];
+    [self takeSnapshot];
+  }
+}
+
+- (void)clearFetchedData
+{
+  [super clearFetchedData];
+  [self removeLiveMap];
+}
+
+- (void)takeSnapshot
+{
+  if (!_snapshotter.isLoading) {
+    [_snapshotter startWithCompletionHandler:^(MKMapSnapshot *snapshot, NSError *error) {
+      if (!error) {
+        UIImage *image = snapshot.image;
+        CGRect finalImageRect = CGRectMake(0, 0, image.size.width, image.size.height);
+        
+        UIGraphicsBeginImageContextWithOptions(image.size, YES, image.scale);
+        [image drawAtPoint:CGPointMake(0, 0)];
+        
+        if (_annotations.count > 0 ) {
+          // Get a standard annotation view pin. Future implementations should use a custom annotation image property.
+          MKAnnotationView *pin = [[MKPinAnnotationView alloc] initWithAnnotation:nil reuseIdentifier:@""];
+          UIImage *pinImage = pin.image;
+          for (id<MKAnnotation>annotation in _annotations)
+          {
+            CGPoint point = [snapshot pointForCoordinate:annotation.coordinate];
+            if (CGRectContainsPoint(finalImageRect, point))
+            {
+              CGPoint pinCenterOffset = pin.centerOffset;
+              point.x -= pin.bounds.size.width / 2.0;
+              point.y -= pin.bounds.size.height / 2.0;
+              point.x += pinCenterOffset.x;
+              point.y += pinCenterOffset.y;
+              [pinImage drawAtPoint:point];
+            }
+          }
+        }
+        
+        UIImage *finalImage = UIGraphicsGetImageFromCurrentImageContext();
+        UIGraphicsEndImageContext();
+        self.image = finalImage;
+      }
+    }];
+  }
+}
+
+- (void)resetSnapshotter
+{
+  if (!_snapshotter.isLoading) {
+    _options.size = self.calculatedSize;
+    _snapshotter = [[MKMapSnapshotter alloc] initWithOptions:_options];
+  }
+}
+
+#pragma mark - Action
+- (void)addLiveMap
+{
+  if (self.isNodeLoaded && !_mapNode) {
+    _mapNode = [[ASDisplayNode alloc]initWithViewBlock:^UIView *{
+      _mapView = [[MKMapView alloc]initWithFrame:CGRectMake(0.0f, 0.0f, self.calculatedSize.width, self.calculatedSize.height)];
+      _mapView.delegate = _mapDelegate;
+      [_mapView setRegion:_options.region];
+      [_mapView addAnnotations:_annotations];
```

Comment:
Hey @appleguy, We already have the setAnnotations method on line 49 so I've extend this to include the above behaviour you mentioned.